pkg/operator/staticpod/controller/backingresource/backing_resource_controller.go

```diff
@@ -121,38 +104,37 @@ func (c BackingResourceController) sync() error {
 		return nil
 	}
 
-	errors := []string{}
 	directResourceResults := resourceapply.ApplyDirectly(c.kubeClient, c.eventRecorder, c.mustTemplateAsset,
 		"manifests/installer-sa.yaml",
 		"manifests/installer-cluster-rolebinding.yaml",
 	)
 
+	errs := []error{}
 	for _, currResult := range directResourceResults {
 		if currResult.Error != nil {
-			errors = append(errors, fmt.Sprintf("%q (%T): %v", currResult.File, currResult.Type, currResult.Error))
+			errs = append(errs, fmt.Errorf("%q (%T): %v", currResult.File, currResult.Type, currResult.Error))
 		}
 	}
+	err = common.NewMultiLineAggregate(errs)
 
-	// No errors, means we succeeded. Reset the state of the failing condition (if exists) as a result.
-	// TODO: This will be replaced by something smarter in near future.
-	if len(errors) == 0 {
-		return c.resetFailingConditionForReason(operatorStatus, resourceVersion, "CreateBackingResourcesError")
+	// update failing condition
+	cond := operatorv1.OperatorCondition{
+		Type:   operatorStatusBackingResourceControllerFailing,
+		Status: operatorv1.ConditionFalse,
 	}
-
-	v1helpers.SetOperatorCondition(&operatorStatus.Conditions, operatorv1.OperatorCondition{
-		Type:    operatorv1.OperatorStatusTypeFailing,
-		Status:  operatorv1.ConditionTrue,
-		Reason:  "CreateBackingResourcesError",
-		Message: strings.Join(errors, ","),
-	})
-
-	if !reflect.DeepEqual(originalOperatorStatus, operatorStatus) {
-		if _, updateError := c.operatorConfigClient.UpdateStatus(resourceVersion, operatorStatus); updateError != nil {
-			glog.Error(updateError)
+	if err != nil {
+		cond.Status = operatorv1.ConditionTrue
+		cond.Reason = "Error"
+		cond.Message = err.Error()
+	}
+	if _, updateError := common.UpdateStatus(c.operatorConfigClient, common.UpdateConditionFn(cond)); updateError != nil {
+		glog.Error(updateError)
```

Comment:
yeah, this is double displaying.  Note the HandleError down below in processNextWorkItem function. Remove these throughout when you return the error.

Reply:
done everywhere